packages/app/src/cli/services/dev/output.ts

```diff
@@ -98,3 +148,26 @@ async function partnersURL(organizationId: string, appId: string) {
     },
   }
 }
+
+function buildPollForDevPreviewMode(apiKey: string, token: string, interval = 10) {
+  return (footerContext: FooterContext) => {
+    const currentIntervalInSeconds = interval
+
+    return new Promise<void>((_resolve, _reject) => {
+      const onPoll = async () => {
+        const app = await fetchAppFromApiKey(apiKey, token)
+        const currentShortcutAction = footerContext.footer?.shortcuts.find((shortcut) => shortcut.key === 'd')
```

Comment:
<img width="1205" alt="Screenshot 2023-08-11 at 10 43 04 AM" src="https://github.com/Shopify/cli/assets/1966327/2f0d57e0-1369-4070-9220-e791c2e5aa35">

We do need this check because if for any reason the shortcut doesn't exist, we should do nothing. The updateShortcut function will also complain if it receives undefined.